ArduCopter/mode_poshold.cpp

```diff
@@ -311,24 +311,28 @@ void ModePosHold::run()
                     brake.angle_max_pitch_cd = fabsf(brake.pitch_cd);
                 } else {
                     // braking angle has started decreasing so re-estimate braking time
-                    brake.timeout_pitch = 1+(uint16_t)(LOOP_RATE_FACTOR*15L*(int32_t)(fabsf(brake.pitch_cd))/(10L*(int32_t)g.poshold_brake_rate));  // the 1.2 (12/10) factor has to be tuned in flight, here it means 120% of the "normal" time.
+                    brake.start_time_pitch_ms = now_ms;
                     brake.time_updated_pitch = true;
                 }
-            }
-
-            // if velocity is very low reduce braking time to 0.5seconds
-            if ((fabsf(vel_fw_cms) <= POSHOLD_SPEED_0) && (brake.timeout_pitch > 50*LOOP_RATE_FACTOR)) {
-                brake.timeout_pitch = 50*LOOP_RATE_FACTOR;
-            }
-
-            // reduce braking timer
-            if (brake.timeout_pitch > 0) {
-                brake.timeout_pitch--;
             } else {
-                // indicate that we are ready to move to Loiter.
-                // Loiter will only actually be engaged once both pitch_mode and pitch_mode are changed to RPMode::BRAKE_READY_TO_LOITER
-                //  logic for engaging loiter is handled below the pitch and pitch mode switch statements
-                pitch_mode = RPMode::BRAKE_READY_TO_LOITER;
+                // scaling factors:
+                // 1.5 times the time taken to level the aircraft in ms
+                // 1000 to convert from seconds to ms
+                // 0.01 to convert angle_max_pitch_cd to degrees
+                const uint32_t brake_timeout_pitch_ms = MIN(POSHOLD_BRAKE_TIME_ESTIMATE_MAX_MS, (1.5 * 1000 * 0.01) * brake.angle_max_pitch_cd / g.poshold_brake_rate_degs);
```

Comment:
same comment here as for roll.  Perhaps we should constrain g.poshold_brake_rate_degs to some acceptable values somewhere